src/components/events/EditModal.jsx

```diff
@@ -0,0 +1,187 @@
+import React, { useState } from "react";
+import axios from "axios";
+import { XIcon } from "../icons/Icons";
+import { ErrorText } from "../form/ErrorText";
+import { LoadingAnimation } from "../icons/LoadingAnimation";
+import { DAYS_ENUM } from "../../utils/common";
+import { GenericInput } from "../form/GenericInput";
+import { IdIcon } from "../icons/Icons";
+import { useUserState } from "../../utils/userState";
+import { DeleteDeal } from "../../server/deal";
+export default function EditModal({ eventData, onDismiss }) {
+    const [user] = useUserState();
+    const [loading, setLoading] = React.useState(false);
+    const { placeId,
+        dayOfWeek,
+        startTime,
+        endTime,
+        title,
+        deals,
+        place,
+        verifiedAt,
+        _id,
+        tags } = eventData
+    const [newDeal, setNewDeal] = useState({})
+    const [error, setError] = useState('')
+    const onClickEditDeal = async () => {
+        try {
+            if(!newDeal) {
+                setError('Please fill fields.')
+                return
+            }
+            await axios.put(`${process.env.REACT_APP_API_URL}/api/deal/edit/${_id}`, {
+                ...newDeal
+            }, {
+                headers: {
+                  Authorization: `${user.token}`,
+                },
+              })
+              window.location.reload()
+        } catch(e) {
+            setError(e)
+            console.log(e)
```

Comment:
nit: rather than just logging error here can we also say `console.log("Error in editing modal:",e)` so we know where the error is originating